### PR TITLE
Clean up minion shutdown

### PR DESCRIPTION
**Pull request by cachedout**
Also cleaner logging for cases where the master could not be contacted.

### CODE REVIEW

**Comment by cachedout** [comment]
This isn't ready yet. Please don't merge this.

File: salt/minion.py [hunk 1]
```diff
@@ -1688,35 +1697,46 @@ def tune_in(self, start=True):
         ping_interval = self.opts.get('ping_interval', 0) * 60
         if ping_interval > 0:
             def ping_master():
-                self._fire_master('ping', 'minion_ping')
+                try:
+                    self._fire_master('ping', 'minion_ping')
+                except Exception:
```
Comment (basepi) [inline salt/minion.py]: We should log the exception, since we're catching all Exceptions here. Otherwise it's hard to figure out why it failed.

**Comment by s0undt3ch** [comment]
Why are you catching keyboard interrupts? The signal handling code handles it, was there really a need for it?

It might just be a matter of calling destroy on the signal handling function?

File: salt/minion.py (hunk 1 above)
Comment (s0undt3ch) [inline salt/minion.py]: Perhaps just add `exc_on_loglevel=logging.DEBUG` ?

**Comment by cachedout** [comment]
@s0undt3ch For whatever reason, that signal handling wasn't being respected by Tornado and I found that catching the KeyboardInterrupt here was necessary to avoid stacktracing.

File: salt/minion.py (hunk 1 above)
Comment (cachedout) [inline salt/minion.py]: Sounds good. Will change. Thanks to both of you.

**Comment by s0undt3ch** [comment]
How can I trigger your stacktrace? Simply by running the minion and trying to stop it with CTRL-C ?

**Comment by cachedout** [comment]
Start a minion and point it to a master which is not up. While it is trying to connect, ctrl-c.

**Comment by s0undt3ch** [comment]
Cool, I'll try that...

**Comment by cachedout** [comment]
@s0undt3ch Were you able to reproduce this?

**Comment by s0undt3ch** [comment]
Didn't had time to go over this, sorry...

**Comment by cachedout** [comment]
@s0undt3ch No problem. Do you oppose this going in or would you prefer to wait?

**Comment by s0undt3ch** [comment]
No, no! Fire away!

**Pull request merged**